actions/gradle/publish/README.md

```diff
@@ -11,3 +11,271 @@ Before you use this publish Action, you will need to configure your Github proje
+                    developerConnection.set("scm:git:ssh://github.com:simpligility/test-java-project.git")
+                    url.set("http://github.com/adamkorcz/test-java-project/tree/main")
+                }
+            }
+        }
+    }
+    repositories {
+        maven {
+            credentials {
+                username = System.getenv("MAVEN_USERNAME")
+                password = System.getenv("MAVEN_PASSWORD")
+            }
+            name = "test-java-project"
+            url = uri("https://s01.oss.sonatype.org/service/local/staging/deploy/maven2/")
+        }
+    }
+}
+
+signing {
+    useGpgCmd()
+    sign(publishing.publications["maven"])
+}
+```
+
+You don't need to configure anything inside that code snippet; Adding them to your `build.gradle.kts` file is enough.
+
+### Add the publish action to your release workflow
+
+Before using the Gradle publish action, you should have a workflow that invokes the Gradle builder. It will look something like this:
+
+```yaml
+name: Publish Gradle with action
+on:
+  - workflow_dispatch
+
+permissions: read-all
+
+jobs:
+  build:
+    permissions:
+      contents: read
+      id-token: write
+      actions: read
+      packages: read
+    uses: AdamKorcz/slsa-github-generator/.github/workflows/builder_gradle_slsa3.yml@make-gradle-publisher-an-action
+    with:
+      rekor-log-public: true
+      artifact-list: build/libs/artifact1-0.1.18.jar,build/libs/artifact-0.1.18-javadoc.jar,build/libs/artifact-0.1.18-sources.jar
+```
+
+To use the Publish action, you need to add another job:
+
+```yaml
+publish:
+    runs-on: ubuntu-latest
+    needs: build
+    permissions:
+      contents: read
+      id-token: write
+      actions: read
+  steps:
+      - name: publish
+        id: publish
+        uses: AdamKorcz/slsa-github-generator/actions/gradle/publish@make-gradle-publisher-an-action
+        with:
+          provenance-download-name: "${{ needs.usetrw.outputs.provenance-download-name }}"
+          provenance-download-sha256: "${{ needs.usetrw.outputs.provenance-download-sha256 }}"
+          target-download-sha256: "${{ needs.usetrw.outputs.target-download-sha256 }}"
+          maven-username: ${{ secrets.OSSRH_USERNAME }}
+          maven-password: ${{ secrets.OSSRH_PASSWORD }}
+          gpg-key-pass: ${{ secrets.GPG_PASSPHRASE }}
+          gpg-private-key: ${{ secrets.GPG_PRIVATE_KEY }}
+          jdk-version: "17"
+```
+
+The variable names of the secrets may be different than how you named the required secrets for publishing to Maven Central. The parameters to `provenance-download-name`, `provenance-download-sha256` and `target-download-sha256` should not be changed.
+
+Once you trigger this workflow, your artifacts and provenance files will be added to a staging repository in Maven Central. You need to close the staging repository and then release:
+
+Closing the staging repository:
```

Comment:
Is closing the staging repo and releasing by hand a common thing Gradle developers do? If not, can we simplify for users (e.g., publish to release rather than staging)

/cc @loosebazooka

Reply:
> Is closing the staging repo and releasing by hand a common thing Gradle developers do?

AFAIK, yes. See this part of Sonatypes guide: https://help.sonatype.com/repomanager2/staging-releases/configuring-your-project-for-deployment#:~:text=Deployment%20with%20the%20Maven%20Deploy%20Plugin,-When%20using%20the&text=If%20the%20staging%20suite%20is,in%20a%20temporary%20staging%20repository.

There is no way we can get around publishing to a staging repository because we are adding more files manually (the provenance files). This is a limitation with Maven Central that they need to ease.

Reply:
Thanks for the info. Resolving then.